packages/journey-manager/src/ui.ts

```diff
@@ -0,0 +1,380 @@
+  <circle cx="15" cy="13" r="1"></circle>
+  <path d="M18 11.03C17.52 8.18 15.04 6 12.05 6c-3.03 0-6.29 2.51-6.03 6.45 2.47-1.01 4.33-3.21 4.86-5.89 1.31 2.63 4 4.44 7.12 4.47"></path>
+</svg>
+`;
+
+const arrowBackIcon = `
+<svg viewBox="0 0 24 24" stroke="none" fill="currentColor">
+  <path d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20z"></path>
+</svg>
+`;
+
+/**
+ * @hidden @internal
+ */
+export class JourneyManagerElement extends HTMLElement {
+  _teardown: (() => void) | null = null;
+  _client: Client | null = null;
+
+  /**
+   * Set Voice Compass client
+   */
+  set client(client: Client) {
+    this._client = client;
+  }
+
+  /**
+   * Set UI configuration
+   */
+  set config(config: UiConfig) {
+    this._teardown?.();
+
+    const shadow = this.attachShadow({ mode: "open" });
+    shadow.innerHTML = `
+<style>${styles(mergeWithDefault(config.theme))}</style>
+<button class="pin">${voiceCompassIcon}</button>
+<div class="drawer">
+  <div class="drawer-content">
+    <div class="drawer-header">
+      <h1>${config.title}</h1>
+      <p>${config.subtitle}</p>
+    </div>
+    <div class="drawer-buttons">
+      <button data-action="previous">
+        ${arrowBackIcon}
+        <span>Previous Screen</span>
+      </button>
+      <button data-action="escalate">
+        ${supportAgentIcon}
+        Escalate to Agent
+      </button>
+    </div>
+    <div class="drawer-footer">
+      <button data-action="close">Close</button>
+    </div>
+  </div>
+</div>
+`;
+
+    let isOpen = false;
+
+    const pinButton = shadow.querySelector(".pin");
+    const drawer = shadow.querySelector(".drawer");
+    const drawerContent = shadow.querySelector(".drawer-content");
+
+    const toggleDrawer = (): void => {
+      isOpen = !isOpen;
+      if (isOpen) {
+        drawer?.classList.add("drawer-open");
+      } else {
+        drawer?.classList.remove("drawer-open");
+      }
+    };
+
+    const handlePinButtonClick = (): void => {
+      toggleDrawer();
+    };
+
+    const handleDrawerClick = (event: Event): void => {
+      const eventTarget = event.target;
+      if (!(eventTarget instanceof HTMLElement)) {
+        return;
+      }
+
+      const action = eventTarget.getAttribute("data-action");
+      if (action != null) {
+        if (action === "close") {
+          toggleDrawer();
+          return;
+        }
+        this.dispatchEvent(
+          new CustomEvent("action", {
+            detail: {
+              action,
+            },
+          }),
+        );
+        return;
+      }
+      if (drawerContent != null && !drawerContent?.contains(eventTarget)) {
+        toggleDrawer();
+      }
+    };
+
+    pinButton?.addEventListener("click", handlePinButtonClick);
+    drawer?.addEventListener("click", handleDrawerClick);
+
+    this._teardown = () => {
+      pinButton?.removeEventListener("click", handlePinButtonClick);
+      drawer?.removeEventListener("click", handleDrawerClick);
+    };
```

Comment:
Is this necessary? Presumably being removed from the DOM cleans up event listeners?

Reply:
I don't think so, how sure would you say you are? I realize this is not too important anyway, just doing the discipline thing :).